pytest_html/plugin.py

```diff
@@ -95,23 +97,30 @@ def __init__(self, logfile):
         self.passed = self.skipped = 0
         self.xfailed = self.xpassed = 0
         self.rerun = 0
+        self.self_contained = self_contained
 
     class TestResult:
 
-        def __init__(self, outcome, report):
+        def __init__(self, outcome, report, self_contained, logfile):
             self.test_id = report.nodeid
             if report.when != 'call':
                 self.test_id = '::'.join([report.nodeid, report.when])
             self.time = getattr(report, 'duration', 0.0)
             self.outcome = outcome
             self.additional_html = []
             self.links_html = []
+            self.self_contained = self_contained
+            self.logfile = logfile
 
-            self.append_log_html(report, self.additional_html)
+            test_index = -1
+            if outcome == 'Rerun':
+                test_index = report.rerun
```

Comment:
See my later comment about possibly even incrementing it to avoid using a negative number:

```python
test_index = hasattr(report, 'rerun') and report.rerun + 1 or 0
```